### ISSUE TITLE
mdx/no-unused-expressions Error when using safe optional operator "?"

### ISSUE THREAD

**Issue by SarantopoulosKon** [mdx-js/eslint-mdx#167]
#####Getting an eslint error when using safe optional operator in my typescript files.

### Subject of the issue
I use eslint-mdx to lint my mdx stories. In my typescript project i use safe optional operator and i get `Expected an assignment or function call and instead saw an expression. (mdx/no-unused-expressions)` error from the eslint mdx plugin. Only workaround i have found is to disable the rule for all ts/tsx files.

### Expected behaviour
It should not show an error

**Comment by JounQin** [comment]
Reproduction is required.

**Comment by SarantopoulosKon** [comment]
I made a [reproduction repo here](https://github.com/SarantopoulosKon/mdx-safe-optional-fn-call)

In `App.tsx` in the `ExampleComponent` you can see i'm using optional chaining with function call and getting an error. The error occurs only for safe optional function calling. And according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#Optional_chaining_with_function_calls), "you can use optional chaining when attempting to call a method which may not exist."


Line(20) that shows error:

```
props.onChange?.(event);
```

**Comment by JounQin** [comment]
`mdx/no-unused-expressions` is calling `no-unused-expressions` from `ESLint`:

https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/src/rules/no-unused-expressions.ts#L18

And when I changed your config to:

```js
module.exports = {
  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
  extends: [
    "plugin:react/recommended", // Uses the recommended rules from @eslint-plugin-react
    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
    "plugin:prettier/recommended" // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
    //   'plugin:mdx/recommended', // ESLint Parser/Plugin for MDX
  ],
  rules: {
    "no-unused-expressions": 2
  }
};
```

There still will be an error:

```log
20:3   error    Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

So you should post issue to `ESLint` itself instead.

---

Besides, you can use `overrides` for `.mdx`:

```js
module.exports = {
  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
  extends: [
    "plugin:react/recommended", // Uses the recommended rules from @eslint-plugin-react
    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
    "plugin:prettier/recommended" // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  overrides: [
    {
      files: "*.mdx",
      extends: "plugin:mdx/recommended" // ESLint Parser/Plugin for MDX
    }
  ]
};
```
